### DISCUSSION TITLE
How to obtain results similar to Photoshop's generative ai with Krita plugin

### DISCUSSION THREAD

**Discussion by lostsole76** [Acly/krita-ai-diffusion#1230] [Q&A]
Hi, I've been doing an extensive project with Photoshop's AI with landscapes and adding assets, flowers, trees, etc. I just discovered Krita plugin. However, after trying all manner of models and settings, I can't seem to get any results that are quite as realistic or that blend as well as what Photoshop has done. Can anyone point me to any information about how this could be done, what models or settings? Is it even possible with Krita plugin to get the same sort of results? Outside of my project, I found the plugin to be amazing! Thank you for any help!

**Comment by minsky91** [comment]
In my experience, I was able to generate all kinds of realistic imagery with the Krita Ai plugin, including objects found in nature (trees, flowers, animals etc.). Organic blending of additionally generated objects requires some understanding of the plugin's specifics though, as the Krita plugin's approach, based on ComfyUI as its backend and Stable Diffusion in general, might differ somewhat from the one used in the generative fill in Photoshop. And there is indeed a plenty of parameters involved in using the plugin that might feel a bit overwhelming.

If you shed a little more light on how you are trying to achieve your goal with Krita AI, together with samples of your successful results in Photoshop perhaps, it might be a little easier to answer to your question.

Note that we have now a brand new Discord-based Kriita AI diffusion user group, you might find more people there who have an extensive experience with the plugin (and are willing to advise) than here . Join us at https://discord.com/invite/pWyzHfHHhU  and post your question in the #general channel!

**Comment by lostsole76** [comment]
![PS GAI Example](https://github.com/user-attachments/assets/3426a93c-a104-467c-8238-b269e6765aa6)

Thanks so much for your response! 

This file is part Photoshop generative fill and part real photography.

In the attached file, the pine trees are from PS GF, but trust they match the rest of the photo. Some of the granite with lichen is PSGF, and some is real. It's hard to tell which is and which is not real. 

The flowers are PS GF and the tall grass is also PS GF, but matches exactly the grass that was there, except the new AI grass is in better shape. 

So, using Krita plugin, and again, trying many experiments with settings and models, and depth of effect sliders, I was unable to duplicate this level of PSGF photorealism. I got some pretty results, but everything had a 3D animation look which won't work with the project. 

I'm willing to compensate someone, if not yourself, that has the knowledge and skill to help me get Krita to do the same as PS GF, if possible? 

I'm also looking for a professional level AI landscape artist to help with my project in a paid freelancer capacity, if you happen to know anyone in your network.

Thanks again for any help!

**Reply by minsky91** [reply]
As a way of promoting Krita AI, I feel motivated enough to take on the challenge, gratis, given available time. If you want a more robust arrangement, my suggestion is to try posting this in the said Discord group / channel, which I now consider 'my network' (as small as it is at the moment - 200 members). Another potentially suitable forum for such a request is Stable Diffusion subreddit (https://www.reddit.com/r/StableDiffusion/), I've seen messages of the kind being posted there from time to time.

"In the attached file, the pine trees are ..." - do you mean you posted more than one image? I don't see any with pines. Also, looking at the image featuring granite with lichen, I get an impression of the top part (the green growth with yellow flowers) suffering from an excessive jpeg compression, though I might be wrong. Social media platforms have a habit of doing that with user-posted images. Could you post the original/base image where you want an AI-generated fragment to be incorporated, in an uncompressed format and target resolution?

**Reply by lostsole76** [reply]
I appreciate it, but I can certainly pay you. I have other artists and programmers working on my project I pay every week. 

You may wonder if Photoshop is working for my needs, why am I bothering with other options? The reason is, as mentioned, my project is extensive. Over the coming years, I will need to do tens of thousands of AI renders. For now, Adobe is giving informally, unlimited credits if you have a software subscription. I imagine they will put a cap on this at some point and I don't know what sort of quantities they will let us buy beyond the subscription limits in the future. So, I need other options in case they tighten things down. 

On the pine trees, I'm sorry, I meant the little pine branches at the top of the rocks. I've included the original image with low compression (Too big for no compression) and then the image with a mix of Photoshop generative fill. 

I'm hoping that Krita plugin can match the photo-realism of PSGF, and to be able to blend as well as it does, if possible? As mentioned, I'm doing massive numbers of these so I need a good smooth workflow to do this. Oh, and target resolution can be the same as the file. 

Also, I'm not in a rush, so no hurry on anything. I just appreciate the help!

Thanks again and let me know if you decide you would like payment?

![Photoshop Generative Fill Mixed Raw Image](https://github.com/user-attachments/assets/65ef20be-4b0a-4892-94f2-12b0fbf4948a)
![Raw image](https://github.com/user-attachments/assets/8af2346a-1ab0-46c2-91c6-57e9537b0284)

**Reply by minsky91** [reply]
Thank you for the additional material. A quick question before delving any deeper: when trying Krita AI for your project, did you by a chance use an external (existing) ComfyUI server? When using it, I indeed get those plastic-looking (not even a 3D animation look) plants and flowers, see the 1st image attached. When using a managed ComfyUI (that is, installed with the Krita AI plugin), I easily get what can hopefully be called a decent realistic-looking AI-generated fragment (see the 2nd image). Such a failure by the standard server is a bit surprising to me as well - so far I can't remember seeing such a drastic difference in results by two flavors of the backend. This is probably a matter of some spurious SD model that can/should be replaced on the system disk.  

The result in the 2nd image was produced with the Generate (Custom)) function of the plugin, in the Seamless mode, although there are other ways to achieve the same. The selection atop of the boulder to incorporate the AI-generated fragment was made so as to approximate what I believe you could have used in your image.

<img width="969" alt="Raw image 50%-generated-20240929-222741-0-patches of green grass and flowers growi" src="https://github.com/user-attachments/assets/39bac7ce-e485-40a9-b694-63de3764d84b">
<img width="969" alt="Raw image 50%-generated-20240929-223323-0-patches of green grass and flowers growi" src="https://github.com/user-attachments/assets/6d07215a-9430-4045-a923-72c24758196a">

(testing on half size-images for speed)

**Reply by minsky91** [reply]
For further communication, can you contact me over Discord (I am minsky91 there) or over reddit (my nic there u/Ok-Vacation5730)? So as not to burden github with our deliberation any longer.

**Reply by lostsole76** [reply]
How would you feel about good old email? I find Discord a bit confusing and my files should be proprietary anyway, so Reddit probably isn't a good choice. If email is okay, how can I private message you my email? To answer your question, I did try Interstice, but I don't recall getting better results. Still, I will try it again. I was hoping to figure it out with a local server setup since I have an RTX 3090, but non local would be okay if it works. At the end of the day, I just don't want a credit limitation for my work. Thank you so far!

**Reply by minsky91** [reply]
emailing is perfectly fine, just send me your address over any of the two platforms I mentioned, by means of a singular contact. (since I feel neither of us wants to share the address here)   

Interstice hosts, as I imagine, a vanilla ComfyUI with the latest version installed, which is why you were/are getting bad results. I don't think you will solve the issue with Interstice without assistance of Acly.  The version supplied with the plugin, on the other hand, is tailored to its work. Do try a local server setup on your RTX 3090, it's the best way to proceed. Will sort out Interstice later on.

**Reply by lostsole76** [reply]
Sent a request on Discord! Thanks! I prefer old school, just plain old simple email. Well, old school to me was writing letters by hand in cursive. Lol.

**Reply by Acly** [reply]
To clarify, there is no difference in output between local, managed or interstice - all produce identical images (with same input).

**Reply by lostsole76** [reply]
Thanks Acly. That is what would seem logical to me.

**Reply by minsky91** [reply]
> To clarify, there is no difference in output between local, managed or interstice - all produce identical images (with same input).

Well, having learned something about ComfyUI setups (which can be very complex and often, let's say, inconsistent) in the last months, I would prefer using the "should be" verb instead of "is". ;) After having spent a whole day investigating this issue, I finally got to the bottom of it - it was most likely a mix-up in custom nodes due to numerous successive updates of Comfy under Stability Matrix in the last weeks that went under the radar of the plugin. When generating content of an image selection on my laptop over a local network connection to the said Comfy server (the latest v0.2.2) running on my main PC, the process had been consistently failing with an endless series of messages of the "_patch type not recognized fooocus ...._" type displayed on the console (which I found out only hours later), but the plugin would not react in any way to that and kept producing that lousy patch of plastic grass (see the 1st image of the 2 I posted above) each and every time. In other words, there was no indication of any error in my Krita interface on the laptop.  At the same time, no render problem whatsoever had occurred when generating from absolutely the same input under the same settings on my main PC with the plugin using the managed server.  No problem either in any version of Comfy, local or external, when  generating an entire image - the screwed renderings happened only when inpainting something inside a selection. 

The difference in output between using the local and external comfy servers was so obvious to me that I concluded, perhaps hastily, that any external, non-managed Comfy server install could be at fault here. Now, after having done (for an umptieth time in my short AI experience) a clean reinstall of ComfyUI, reinstalling as well all the needed nodes and models and whatnot, I have it running again properly on all my configurations, no inpainting issues that I can notice. For how long, I don't know.
